hazelcast/src/main/java/com/hazelcast/config/RingbufferConfig.java

```diff
@@ -0,0 +1,322 @@
+     * @return the total number of backups: backupCount plus asyncBackupCount
+     */
+    public int getTotalBackupCount() {
+        return backupCount + asyncBackupCount;
+    }
+
+    /**
+     * Creates a readonly copy of this RingbufferConfig.
+     *
+     * @return the readonly copy.
+     */
+    public RingbufferConfig getAsReadOnly() {
+        return new RingbufferConfigReadonly(this);
+    }
+
+    /**
+     * Gets the retention in seconds.
+     *
+     * @return the retention in seconds. Returns 0 if there is no retention.
+     */
+    public int getRetentionSeconds() {
+        return retentionSeconds;
+    }
+
+    /**
+     * Sets the retention in seconds.
+     *
+     * Retention is the time the ringbuffer is going to retain items before deleting them.
+     *
+     * Retention can also be disabled by setting retentionSeconds to 0. It means that items won't get removed because they
+     * retire. They will only overwrite. This means that when retention is disabled, that after tail did a full loop in the ring
+     * that the size will always be equal to the capacity.
+     *
+     * @param retentionSeconds the retention period in seconds
+     * @return the updated RingbufferConfig
+     * @throws IllegalArgumentException if retentionSeconds smaller than 0.
+     */
+    public RingbufferConfig setRetentionSeconds(int retentionSeconds) {
+        this.retentionSeconds = checkNotNegative(retentionSeconds, "retention can't be smaller than 0");
+        return this;
+    }
+
+    /**
+     * Gets the InMemoryFormat.
+     *
+     * @return the InMemoryFormat.
+     */
+    public InMemoryFormat getInMemoryFormat() {
+        return inMemoryFormat;
+    }
+
+    /**
+     * Sets the InMemoryFormat.
+     *
+     * Setting the InMemoryFormat controls format of storing an item in the ringbuffer:
+     * <ol>
+     *     <li>{@link InMemoryFormat#OBJECT}: the item is stored in deserialized format (so a regular object)</li>
+     *     <li>{@link InMemoryFormat#BINARY}: the item is stored in serialized format (so a is binary blob) </li>
+     * </ol>
+     *
+     * The default is binary. The object InMemoryFormat is useful when:
+     * <ol>
+     *     <li>of the object stored in object format has a smaller footprint than in binary format</li>
+     *     <li>if there are readers using a filter. Since for every filter invocation, the object needs to be available in
+     *     object format.</li>
+     * </ol>
+     *
+     * @param inMemoryFormat the new in memory format.
+     * @return the updated Config.
+     * @throws NullPointerException if inMemoryFormat is null.
+     * @throws IllegalArgumentException if {@link InMemoryFormat#NATIVE} in memory format is selected.
+     */
+    public RingbufferConfig setInMemoryFormat(InMemoryFormat inMemoryFormat) {
+        checkNotNull(inMemoryFormat, "inMemoryFormat can't be null");
+        checkFalse(inMemoryFormat == NATIVE, "InMemoryFormat " + NATIVE + " is not support");
```

Comment:
actually you write, i want to see some tests because;
when user makes wrong config in xml hazelcast will throw invalidConfigurationException only
when user makes wrong config in Config object( lets say 10 backup-count) hazelcast will throw illegalArgumentException..
so exceptions are different.